### ISSUE TITLE
Error: All fetchers failed for spack-stage

### ISSUE THREAD

**Issue by josepedro** [LLNL/spack#17172]
spack fetch -D hpctoolkit
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/git//HPCToolkit/hpctoolkit.git/ac6ae1156e77d35596fea743ed8ae768f7222f19.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/89/89f010078b6cf69c23c27897d686055ab89b198dddf819efb0a4f2c38a0b36e6.tar.bz2
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/d3/d34272e4042c46186ddcd66bd5d980c0ca14ff734444686ccf8131f6ec8b1427.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/66/66db7df8a91979eb576fac91743644da878244cf8ee152f02cd6f5cd7a731689.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/9d/9da50e155df72bce55cb69f51f1dbb4b62d23740fb99f6178bb27f22ebdf8a46.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/86/86e613527e5dba544e73208f42b78b7c022d4fa5a6d5498bf18c8d6f745b91dc.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/e3/e339f51971478d369f8a053a330a190781acb9864cf4c541060f12078948e461.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/30/30306e0c76e0f9f1f0de987cf1c82a5c21e1ce6568b9227f7da5b71cbea86c9d.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/b8/b846aea51cf696c3392a0ae58bef93e2e72f8e7073ca6ad1ed8b01c85871f9c0.tar.xz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/c3/c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/4e/4eb3b8d442b426dc35346235c8733b5ae35ba431690e38c6a8263dce9fcbb402.tar.bz2
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/4d/4dd507e1f5a29e3b87b15321a4d8c74afdc8331433edabf7aeab89b3c405d556
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/24/246508e052c44b6f4e8c2542a71c06cacaa72cd1447ab8d2a542b987bc35ace9
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/ab/ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/b3/b3a7a6221c3dc916085f0d205abf6b8e1ba443d4dd965118da364a1dc1cb3a26.tar.xz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/e6/e6a1b1b589654277ee790cce3734f07876ac4ccfaecbee8afa0b649cf529cc04.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/git//dyninst/dyninst.git/v10.1.0.tar.gz
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/3a/3aa9114485da39cbd9665a0bfe986894a282d5f0882b1dea960a739496620727.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/dd/ddb04774f1e32f0c49751e21b67216ac87852ceb056b75209af2443400636d46.tar.gz
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/25/25a545566cbacaa37ae6222e58f1c48ea4570f53ba991886e2f5ce96e22a23a2.tar.bz2
==> Using cached archive: /home/jose/git/spack/var/spack/cache/_source-cache/archive/ab/ab2633921a5cd38e48797bf5521ad259bdc4b979078034a3b790d7fec5493fab.tar.gz
==> Fetching https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch
##O#- #                                                            
curl: (22) The requested URL returned error: 503 Service Temporarily Unavailable
==> Failed to fetch file from URL: https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch
    URL https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch was not found!
==> Fetching from https://src.fedoraproject.org/rpms/m4/raw/5d147168d4b93f38a4833f5dd1d650ad88af5a8a/f/m4-1.4.18-glibc-change-work-around.patch failed.
==> Error: All fetchers failed for spack-stage-6oa8hrid

**Comment by mwkrentel** [comment]
See #17030 for why this keeps failing.

Anyway, if your real goal is to install hpctoolkit (or anything else
that depends on m4), then there's a simple workaround.  Use m4 1.4.17.
M4 is just a build tool and 1.4.17 will work as well as 1.4.18.
1.4.17 doesn't require fetching the remote patch (what really fails).

```
spack install hpctoolkit ... ^m4@1.4.17
```

Better yet, add it to `packages.yaml`.

**Comment by alalazo** [comment]
Duplicate of #17030

**Comment by alalazo** [comment]
@josepedro Closing as duplicate, feel free to reopen if you think there are other issues to be considered.
